### PR TITLE
Staff of the Storyteller support

### PR DESCRIPTION
**Pull request by tool4ever**
I had to move the LKI into the effect because of "one or more creature tokens":
otherwise it would still trigger if _Myrkul_ created enchantment token while you had _Opalescence_ in play.

This way because GameAction has logic to reuse it no new objects get created from that.

Also ``setTriggeringObjects`` is only future-proofing.

### CODE REVIEW

File: forge-game/src/main/java/forge/game/trigger/TriggerType.java [hunk 1]
```diff
@@ -116,6 +116,7 @@ public enum TriggerType {
     Taps(TriggerTaps.class),
     TapsForMana(TriggerTapsForMana.class),
     TokenCreated(TriggerTokenCreated.class),
+    TokenCreatedOnce(TriggerTokenCreated.class),
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/trigger/TriggerType.java:119]: imo would be better if Once uses its own Trigger class

File: forge-gui/res/cardsfolder/a/akim_the_soaring_wind.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ ManaCost:2 U R W
 Types:Legendary Creature Bird Dinosaur
 PT:3/4
 K:Flying
-T:Mode$ TokenCreated | ValidPlayer$ You | OnlyFirst$ True | TriggerZones$ Battlefield | Execute$ TrigToken | TriggerDescription$ Whenever you create one or more tokens for the first time each turn, create a 1/1 white Bird creature token with flying.
+T:Mode$ TokenCreated | ValidToken$ Card.YouOwn | OnlyFirst$ True | TriggerZones$ Battlefield | Execute$ TrigToken | TriggerDescription$ Whenever you create one or more tokens for the first time each turn, create a 1/1 white Bird creature token with flying.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/a/akim_the_soaring_wind.txt:6]: imo this should be a TokenCreatedOnce trigger

**Comment by Hanmac** [comment]
is there a corner case where a Player creates a token, but isn't the owner?

because i liked the ValidPlayer trigger more 🤔

**Comment by tool4ever** [comment]
I don't think so:
> 111.2. The player who creates a token is its owner.

File: forge-gui/res/cardsfolder/a/akim_the_soaring_wind.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/a/akim_the_soaring_wind.txt:6]: The thing is TokenCreatedOnce might contain tokens with different owners
So we'd need to build a `Map<Player, Boolean> firstTimers` and then check that in the trigger

This feels like unnecessary overhead?

Comment (Hanmac) [reply forge-gui/res/cardsfolder/a/akim_the_soaring_wind.txt:6]: Or: make a List of Players that have their token created for the first time imo List would be better than map?
and then use `FirstTime$ You`

**Comment by tool4ever** [comment]
@Hanmac 
for cards like  _Curse of the Swine_ that make tokens with RepeatEach + ChangeZoneTable$ do we need to couple it with triggerChangesZoneAll so in case same controller gets targeted Staff doesn't trigger that many times? 🤔

**Comment by tool4ever** [comment]
word "Create" used once -> single event

Should be handled

**Comment by Agetian** [comment]
Looks good to me. Does anything else need doing here?

**Comment by tool4ever** [comment]
I think it's ready

**Pull request merged**